server/src/main/java/org/opensearch/index/store/remote/file/OnDemandBlockSnapshotIndexInput.java

```diff
@@ -133,10 +137,12 @@ protected OnDemandBlockSnapshotIndexInput buildSlice(String sliceDescription, lo
 
     @Override
     protected IndexInput fetchBlock(int blockId) throws IOException {
-        final String blockFileName = fileName + "." + blockId;
+        logger.trace("fetchBlock called with blockId -> {}", blockId);
```

Comment:
Have changed the unit tests so it shouldn't have much impact.

Change was necessary as in the listAll() of Composite Directory we are skipping the block files and hence needed a more verbose way of identifying the block files, the earlier one with just a dot in the name didn't help much in that case.

Reply:
I think process restart should be fine. But yeah, if there is a version upgrade then we might end up discarding the older downloaded files for searchable snapshot use case.

Reply:
Ya I meant restart as part of version upgrades. But looks like today `FileCache` loads existing files as closed `IndexInput` so it will again recreate a new IndexInput and put that in `FileCache`. So with the name changes, the new version will look for block file with new name and will not match any existing block files on the node and will end up evicting those eventually. So don't see any issue for now. But worth adding a version upgrade test for verification.